### PR TITLE
SITE-1118 Implement appropriate date/time format

### PR DESCRIPTION
**Pull request by lukagabric**
SITE-1118 Based on question type, depending if it is date time or date only, use and submit only the necessary calendar components with the task response.

### CODE REVIEW

File: ResearchKit/Common/ORKAnswerFormat.m [hunk 1]
```diff
@@ -1405,9 +1405,22 @@ - (NSDate *)dateFromString:(NSString *)string {
     return [dfm dateFromString:string];
 }
 
-- (NSDate *)pickerDefaultDate {
-    return (self.defaultDate ? : [NSDate date]);
+- (NSDate *)pickerDefaultDate
```
Comment (diederikh) [inline ResearchKit/Common/ORKAnswerFormat.m:1408]: Can this not be done in `MDORKQuestionStepViewController.answerCellForTableView:`?

**Review by diederikh — CHANGES_REQUESTED** [review]

File: ResearchKit/Common/ORKAnswerFormat.m (hunk 1 above)
Comment (lukagabric) [reply ResearchKit/Common/ORKAnswerFormat.m:1408]: This is a private getter.

Comment (diederikh) [reply ResearchKit/Common/ORKAnswerFormat.m:1408]: You can expose the getter via a subclass. We do this all the time for other RK customizations.

Comment (lukagabric) [reply ResearchKit/Common/ORKAnswerFormat.m:1408]: I did not mean we cannot expose it, rather it is not worth it changing private implementation, it is easier to change in RK. See this conversation for reference: https://github.com/Medable/ResearchKit/pull/29

Comment (lukagabric) [reply ResearchKit/Common/ORKAnswerFormat.m:1408]: @diederikh I think our conversation from the watercooler resolve this?

**Review by diederikh — APPROVED** [review]

**Pull request merged**